Add Operations profile to develop

umm0 opened the pull request:
Add Operations profile to SPDX 3

umm0 committed:
rebase opeartions to develop

Signed-off-by: Ummo Schwarting <uschwarting@deloitte.de>

umm0 committed:
add Operations tab to Drawio after reset - broken after rebase, to be fixed

Signed-off-by: Ummo Schwarting <uschwarting@deloitte.de>

mkurzman committed:
Create Operations.md takeover

taken over from https://github.com/boschglobal/spdx-3-model/blob/mkurzman_20250313/model/Operations/Operations.md

Signed-off-by: Marcel Kurzmann <marcel.kurzmann@bosch.com>

mkurzman committed:
Update business and technical operations and smaller typos.

Signed-off-by: Marcel Kurzmann <marcel.kurzmann@bosch.com>

mkurzman committed:
Added assessmentRelationship to core - colourcoded

Signed-off-by: Marcel Kurzmann <marcel.kurzmann@bosch.com>

mkurzman committed:
OperationsAssessmentRelationship and Project Classes added

The general class "OperationsAssessmentRelationship based on Core/AssessmentReleationship contains all properties that OperationsAssessments have in common.
The Project class based on Core/Organization contains project specific properties.

Signed-off-by: Marcel Kurzmann <marcel.kurzmann@bosch.com>

mkurzman committed:
Create AssessmentRelationship.md

Signed-off-by: Marcel Kurzmann <marcel.kurzmann@bosch.com>

mkurzman committed:
Create Project.md

Signed-off-by: Marcel Kurzmann <marcel.kurzmann@bosch.com>

mkurzman committed:
Create SCRMConcept.md

Signed-off-by: Marcel Kurzmann <marcel.kurzmann@bosch.com>

mkurzman committed:
Create ApplicationType.md

Signed-off-by: Marcel Kurzmann <marcel.kurzmann@bosch.com>

mkurzman committed:
Create DistributionMethodType

Signed-off-by: Marcel Kurzmann <marcel.kurzmann@bosch.com>

mkurzman committed:
Update Project.md

Signed-off-by: Marcel Kurzmann <marcel.kurzmann@bosch.com>

mkurzman committed:
Create ProjectTitle.md

Signed-off-by: Marcel Kurzmann <marcel.kurzmann@bosch.com>

mkurzman committed:
Create ProjectInformation.md

Signed-off-by: Marcel Kurzmann <marcel.kurzmann@bosch.com>

mkurzman committed:
Create ProjectContract.md

Signed-off-by: Marcel Kurzmann <marcel.kurzmann@bosch.com>

mkurzman committed:
Create ProjectSponsor.md

Signed-off-by: Marcel Kurzmann <marcel.kurzmann@bosch.com>

mkurzman committed:
Create ProjectOwner.md

Signed-off-by: Marcel Kurzmann <marcel.kurzmann@bosch.com>

mkurzman committed:
Rename ProjectContract.md to projectContract.md

Signed-off-by: Marcel Kurzmann <marcel.kurzmann@bosch.com>

mkurzman committed:
Rename ProjectInformation.md to projectInformation.md

Signed-off-by: Marcel Kurzmann <marcel.kurzmann@bosch.com>

mkurzman committed:
Rename ProjectOwner.md to projectOwner.md

Signed-off-by: Marcel Kurzmann <marcel.kurzmann@bosch.com>

mkurzman committed:
Rename ProjectSponsor.md to projectSponsor.md

Signed-off-by: Marcel Kurzmann <marcel.kurzmann@bosch.com>

mkurzman committed:
Rename ProjectTitle.md to projectTitle.md

Signed-off-by: Marcel Kurzmann <marcel.kurzmann@bosch.com>

mkurzman committed:
Rename SCRMConcept.md to scrmConcept.md

Signed-off-by: Marcel Kurzmann <marcel.kurzmann@bosch.com>

mkurzman committed:
Update Project.md

Signed-off-by: Marcel Kurzmann <marcel.kurzmann@bosch.com>

mkurzman committed:
applicationType created based on former draft

Signed-off-by: Marcel Kurzmann <marcel.kurzmann@bosch.com>

mkurzman committed:
distributionMethod.md created based on former draft

Signed-off-by: Marcel Kurzmann <marcel.kurzmann@bosch.com>

mkurzman committed:
complianceBundleStorage.md based on former draft of fossComplianceBundleStorage

Signed-off-by: Marcel Kurzmann <marcel.kurzmann@bosch.com>

mkurzman committed:
Create projectStartTime.md

Signed-off-by: Marcel Kurzmann <marcel.kurzmann@bosch.com>

mkurzman committed:
Create projectEndTime.md

Signed-off-by: Marcel Kurzmann <marcel.kurzmann@bosch.com>

mkurzman committed:
Create OperationsAssessmentRelationship

Signed-off-by: Marcel Kurzmann <marcel.kurzmann@bosch.com>

mkurzman committed:
Update OperationsAssessmentRelationship

Signed-off-by: Marcel Kurzmann <marcel.kurzmann@bosch.com>

mkurzman committed:
Create operationsAssessmentComment.md

Signed-off-by: Marcel Kurzmann <marcel.kurzmann@bosch.com>

mkurzman committed:
Update OperationsAssessmentRelationship

Signed-off-by: Marcel Kurzmann <marcel.kurzmann@bosch.com>

mkurzman committed:
Create assessedProject.md

Signed-off-by: Marcel Kurzmann <marcel.kurzmann@bosch.com>

mkurzman committed:
some fixes and typos and consistency with the model-files.

Signed-off-by: Marcel Kurzmann <marcel.kurzmann@bosch.com>

mkurzman committed:
Bugfix "Properties"

Signed-off-by: Marcel Kurzmann <marcel.kurzmann@bosch.com>

mkurzman committed:
Create assessedElement.md

Signed-off-by: Marcel Kurzmann <marcel.kurzmann@bosch.com>

mkurzman committed:
Create publishedTime.md

Signed-off-by: Marcel Kurzmann <marcel.kurzmann@bosch.com>

mkurzman committed:
Create modifiedTime.md

Signed-off-by: Marcel Kurzmann <marcel.kurzmann@bosch.com>

mkurzman committed:
Create withdrawnTime.md

Signed-off-by: Marcel Kurzmann <marcel.kurzmann@bosch.com>

mkurzman committed:
Rename OperationsAssessmentRelationship to OperationsAssessmentRelationship.md

Signed-off-by: Marcel Kurzmann <marcel.kurzmann@bosch.com>

mkurzman committed:
Update projectTitle.md

Signed-off-by: Marcel Kurzmann <marcel.kurzmann@bosch.com>

mkurzman committed:
Update model/Core/Classes/AssessmentRelationship.md

Co-authored-by: Arthit Suriyawongkul <arthit@gmail.com>
Signed-off-by: Marcel Kurzmann <marcel.kurzmann@bosch.com>

mkurzman committed:
Update model/Core/Classes/AssessmentRelationship.md

Co-authored-by: Arthit Suriyawongkul <arthit@gmail.com>
Signed-off-by: Marcel Kurzmann <marcel.kurzmann@bosch.com>

mkurzman committed:
Update model/Operations/Classes/Project.md

Co-authored-by: Arthit Suriyawongkul <arthit@gmail.com>
Signed-off-by: Marcel Kurzmann <marcel.kurzmann@bosch.com>

mkurzman committed:
Update Project.md

Signed-off-by: Marcel Kurzmann <marcel.kurzmann@bosch.com>

mkurzman committed:
Update Project.md

Signed-off-by: Marcel Kurzmann <marcel.kurzmann@bosch.com>

mkurzman committed:
Update model/Operations/Classes/Project.md

Co-authored-by: Arthit Suriyawongkul <arthit@gmail.com>
Signed-off-by: Marcel Kurzmann <marcel.kurzmann@bosch.com>

mkurzman committed:
Update model/Operations/Classes/Project.md

Co-authored-by: Arthit Suriyawongkul <arthit@gmail.com>
Signed-off-by: Marcel Kurzmann <marcel.kurzmann@bosch.com>

mkurzman committed:
Update model/Operations/Properties/applicationType.md

Co-authored-by: Arthit Suriyawongkul <arthit@gmail.com>
Signed-off-by: Marcel Kurzmann <marcel.kurzmann@bosch.com>

mkurzman committed:
Update model/Operations/Properties/applicationType.md

Co-authored-by: Arthit Suriyawongkul <arthit@gmail.com>
Signed-off-by: Marcel Kurzmann <marcel.kurzmann@bosch.com>

mkurzman committed:
Update model/Operations/Properties/distributionMethod.md

Co-authored-by: Arthit Suriyawongkul <arthit@gmail.com>
Signed-off-by: Marcel Kurzmann <marcel.kurzmann@bosch.com>

mkurzman committed:
Update model/Operations/Properties/distributionMethod.md

Co-authored-by: Arthit Suriyawongkul <arthit@gmail.com>
Signed-off-by: Marcel Kurzmann <marcel.kurzmann@bosch.com>

mkurzman committed:
Update model/Operations/Properties/scrmConcept.md

Co-authored-by: Arthit Suriyawongkul <arthit@gmail.com>
Signed-off-by: Marcel Kurzmann <marcel.kurzmann@bosch.com>

mkurzman committed:
Update model/Operations/Vocabularies/ApplicationType.md

Co-authored-by: Arthit Suriyawongkul <arthit@gmail.com>
Signed-off-by: Marcel Kurzmann <marcel.kurzmann@bosch.com>

mkurzman committed:
Rename DistributionMethodType to DistributionMethodType.md

Signed-off-by: Marcel Kurzmann <marcel.kurzmann@bosch.com>

mkurzman committed:
Update publishedTime.md

Signed-off-by: Marcel Kurzmann <marcel.kurzmann@bosch.com>

mkurzman committed:
Update withdrawnTime.md

Signed-off-by: Marcel Kurzmann <marcel.kurzmann@bosch.com>

mkurzman committed:
Update modifiedTime.md

Signed-off-by: Marcel Kurzmann <marcel.kurzmann@bosch.com>

mkurzman committed:
Update projectInformation.md

Signed-off-by: Marcel Kurzmann <marcel.kurzmann@bosch.com>

mkurzman committed:
Update DistributionMethodType.md

Signed-off-by: Marcel Kurzmann <marcel.kurzmann@bosch.com>

mkurzman committed:
Update Project.md

Signed-off-by: Marcel Kurzmann <marcel.kurzmann@bosch.com>

mkurzman committed:
Delete model/Operations/Properties/projectInformation.md

Core/description can be used instead

Signed-off-by: Marcel Kurzmann <marcel.kurzmann@bosch.com>

mkurzman committed:
Update projectContract.md

Signed-off-by: Marcel Kurzmann <marcel.kurzmann@bosch.com>

mkurzman committed:
Update projectOwner.md

Signed-off-by: Marcel Kurzmann <marcel.kurzmann@bosch.com>

mkurzman committed:
Update projectSponsor.md

Signed-off-by: Marcel Kurzmann <marcel.kurzmann@bosch.com>

mkurzman committed:
Update scrmConcept.md

Signed-off-by: Marcel Kurzmann <marcel.kurzmann@bosch.com>

mkurzman committed:
Update projectOwner.md

Signed-off-by: Marcel Kurzmann <marcel.kurzmann@bosch.com>

mkurzman committed:
Update projectSponsor.md

Signed-off-by: Marcel Kurzmann <marcel.kurzmann@bosch.com>

mkurzman committed:
Update operationsAssessmentComment.md

Signed-off-by: Marcel Kurzmann <marcel.kurzmann@bosch.com>

mkurzman committed:
Update complianceBundleStorage.md

Signed-off-by: Marcel Kurzmann <marcel.kurzmann@bosch.com>

mkurzman committed:
Changed based on review on 8.8.2025

Signed-off-by: Marcel Kurzmann <marcel.kurzmann@bosch.com>

mkurzman committed:
Switched to Bundle based on review.

Signed-off-by: Marcel Kurzmann <marcel.kurzmann@bosch.com>

mkurzman committed:
Change Project ancestor to Core/Bundle and deleted the Assessment Class inheritance

Signed-off-by: Marcel Kurzmann <marcel.kurzmann@bosch.com>

mkurzman committed:
Delete model/Core/Classes/AssessmentRelationship.md

Signed-off-by: Marcel Kurzmann <marcel.kurzmann@bosch.com>

mkurzman committed:
Delete model/Core/Properties/withdrawnTime.md

Signed-off-by: Marcel Kurzmann <marcel.kurzmann@bosch.com>

mkurzman committed:
Delete model/Core/Properties/modifiedTime.md

Signed-off-by: Marcel Kurzmann <marcel.kurzmann@bosch.com>

mkurzman committed:
Delete model/Core/Properties/publishedTime.md

Signed-off-by: Marcel Kurzmann <marcel.kurzmann@bosch.com>

mkurzman committed:
Delete model/Core/Properties/assessedElement.md

Signed-off-by: Marcel Kurzmann <marcel.kurzmann@bosch.com>

mkurzman committed:
Delete model/Operations/Classes/OperationsAssessmentRelationship.md

Will be replaced by suggestion from Ummo

Signed-off-by: Marcel Kurzmann <marcel.kurzmann@bosch.com>

mkurzman committed:
Update Project.md

deleted properties that can be covered by Bundle/context

Signed-off-by: Marcel Kurzmann <marcel.kurzmann@bosch.com>

mkurzman committed:
Delete model/Operations/Properties/scrmConcept.md

Signed-off-by: Marcel Kurzmann <marcel.kurzmann@bosch.com>

mkurzman committed:
Delete model/Operations/Properties/operationsAssessmentComment.md

Signed-off-by: Marcel Kurzmann <marcel.kurzmann@bosch.com>

mkurzman committed:
Delete model/Operations/Properties/distributionMethod.md

Signed-off-by: Marcel Kurzmann <marcel.kurzmann@bosch.com>

mkurzman committed:
Delete model/Operations/Properties/complianceBundleStorage.md

Signed-off-by: Marcel Kurzmann <marcel.kurzmann@bosch.com>

mkurzman committed:
Delete model/Operations/Properties/applicationType.md

Signed-off-by: Marcel Kurzmann <marcel.kurzmann@bosch.com>

mkurzman committed:
Delete model/Operations/Vocabularies/DistributionMethodType.md

Signed-off-by: Marcel Kurzmann <marcel.kurzmann@bosch.com>

mkurzman committed:
Delete model/Operations/Vocabularies/ApplicationType.md

Signed-off-by: Marcel Kurzmann <marcel.kurzmann@bosch.com>

mkurzman committed:
Delete model/Operations/Properties/assessedProject.md

Signed-off-by: Marcel Kurzmann <marcel.kurzmann@bosch.com>

mkurzman committed:
Update model/Operations/Classes/Project.md

Co-authored-by: stevenc-stb <steven@smarttalkbeacon.com>
Signed-off-by: Marcel Kurzmann <marcel.kurzmann@bosch.com>

mkurzman committed:
Update model/Operations/Classes/Project.md

Co-authored-by: stevenc-stb <steven@smarttalkbeacon.com>
Signed-off-by: Marcel Kurzmann <marcel.kurzmann@bosch.com>

umm0 committed:
operations: draft for export control classification

Signed-off-by: Ummo Schwarting <uschwarting@deloitte.de>

umm0 committed:
Update model/Operations/Classes/ExportControlClassification.md

Co-authored-by: stevenc-stb <steven@smarttalkbeacon.com>
Signed-off-by: umm0 <uschwarting@deloitte.de>

umm0 committed:
Update model/Operations/Properties/exportingCountry.md

Co-authored-by: stevenc-stb <steven@smarttalkbeacon.com>
Signed-off-by: umm0 <uschwarting@deloitte.de>

umm0 committed:
rebase profile-operations to develop

Signed-off-by: Ummo Schwarting <uschwarting@deloitte.de>

umm0 committed:
update drawio with recent -develop- state and update with Operation tab

Signed-off-by: Ummo Schwarting <uschwarting@deloitte.de>

umm0 committed:
add property assessmentResult

Signed-off-by: Ummo Schwarting <uschwarting@deloitte.de>

umm0 committed:
fix naming mismatch for exportControlSpecification

Signed-off-by: Ummo Schwarting <uschwarting@deloitte.de>

umm0 committed:
fix syntax in ExportControlClassificationAssessment.md

Signed-off-by: Ummo Schwarting <uschwarting@deloitte.de>

umm0 committed:
remove minimum requirement for assesor bc this might clash with some implementations

Signed-off-by: Ummo Schwarting <uschwarting@deloitte.de>

umm0 committed:
Change ExportControlClassificationAssessmen ancestry to Artifact

Signed-off-by: Ummo Schwarting <uschwarting@deloitte.de>

umm0 committed:
fix operations drawio after rebase

Signed-off-by: Ummo Schwarting <uschwarting@deloitte.de>

umm0 committed:
correct ExportControlClassificationAssessment parent in drawio

Signed-off-by: Ummo Schwarting <uschwarting@deloitte.de>

umm0 commented:
Hi @kestewart , this is the pull request to add the Operations profile.

There is probably some renaming to do.
We would also like to gather input on the proposal.

Currently set up as draft. It would be great if you could help set a deadline, when we should finalize and merge.

Best wishes
Ummo

stevenc-stb commented on model/Operations/Properties/exportControlSpecification.md:
```diff
@@ -0,0 +1,17 @@
+
+- name: exportControlSpecification
+- Nature: DataProperty
+- Range: /Core/ExternalIdentifier
```
This could be a /Core/Specification.  but a ExternalIdentifier works too.

stevenc-stb replied:
```suggestion
- Range: /Core/Specification
```

stevenc-stb commented on model/Operations/Classes/ExportControlClassification.md:
```diff
@@ -0,0 +1,37 @@
+  - minCount: 1
+  - maxCount: 1
+- exportControlSpecification
+  - type: /Core/ExternalIdentifier
```
This could be a[ /Core/Specification. ](https://spdx.github.io/spdx-spec/v3.1-dev/model/Core/Classes/Specification/) but a ExternalIdentifier works too.

stevenc-stb replied:
```suggestion
  - type: /Core/Specification
```

stevenc-stb reviewed (changes requested):
This could be a /Core/Specification.

umm0 committed:
Update model/Operations/Classes/ExportControlClassification.md

Co-authored-by: stevenc-stb <steven@smarttalkbeacon.com>
Signed-off-by: Ummo Schwarting <uschwarting@deloitte.de>

umm0 committed:
Update model/Operations/Properties/exportControlSpecification.md

Co-authored-by: stevenc-stb <steven@smarttalkbeacon.com>
Signed-off-by: Ummo Schwarting <uschwarting@deloitte.de>

umm0 commented:
Change from draft state to review decided in Operations meeting Dec 12 2025

bact commented on model/Operations/Properties/exportControlClassificationResult.md:
```diff
@@ -0,0 +1,18 @@
+
+# exportControlClassificationResult
+
+
```
```suggestion

```

bact commented on model/Operations/Operations.md:
```diff
@@ -0,0 +1,22 @@
+
+## Summary
+
+The Operations Profile defines fields for describing the business context of the software that cannot (or not yet) be directly extracted from the source repository. Operations include technical operations and business operations.
```
```suggestion
The Operations profile defines fields for describing the business context of the software that cannot (or not yet) be directly extracted from the source repository. Operations include technical operations and business operations.
```

Use lowercase "profile", according to https://github.com/spdx/spdx-spec/issues/1250

bact commented on model/Operations/Operations.md:
```diff
@@ -0,0 +1,22 @@
+
+## Description
+
+The intention of the Operations Profile is to provide a common base rather for alignment of tool and infrastructure interfaces and development than for exchanging the data between organizations along the software supply chain. In a first phase it shall focus on business operations. The managed information is expected to be mainly kept within the organizations but in some cases might also be neces ...
```
```suggestion
The intention of the Operations profile is to provide a common base rather for alignment of tool and infrastructure interfaces and development than for exchanging the data between organizations along the software supply chain. In a first phase it shall focus on business operations. The managed information is expected to be mainly kept within the organizations but in some cases might also be necessary input for processing and determining relevant parameters for the later exchange in the supply chain (e.g. Export Control Classification Number).
```

bact commented on model/Operations/Operations.md:
```diff
@@ -0,0 +1,22 @@
+
+The information about an Element shall describe facts about the distribution and business context of the Element brought into the market, that could consist of several deliverables. The relation between Element and Deliverable targets can be one to many.
+
+The information about Deliverables shall provide more detailed information about the software item already in early phases of the product lifecycle, where the content may still be a planning or intended value that can be replaced and updated by real values in further phases (e.g. used programming language, used frameworks, ...)
```
```suggestion
The information about deliverables shall provide more detailed information about the software item already in early phases of the product lifecycle, where the content may still be a planning or intended value that can be replaced and updated by real values in further phases (e.g. used programming language, used frameworks).
```

I don't see other instance of "deliverable" with uppercase at first letter.

bact commented on model/Operations/Properties/exportControlClassificationResult.md:
```diff
@@ -0,0 +1,18 @@
+
+## Summary
+
+Provides the result of an export control assessment
```
```suggestion
Provides the result of an export control assessment.
```

bact commented on model/Operations/Properties/exportControlClassificationResult.md:
```diff
@@ -0,0 +1,18 @@
+
+## Description
+
+Provides the result of an export control assessment
```
```suggestion
Provides the result of an export control assessment.
```

bact commented on model/Operations/Operations.md:
```diff
@@ -0,0 +1,22 @@
+
+The information about Deliverables shall provide more detailed information about the software item already in early phases of the product lifecycle, where the content may still be a planning or intended value that can be replaced and updated by real values in further phases (e.g. used programming language, used frameworks, ...)
+
+The Operations Assessment descriptions are patterns for business relevant assessments like an ExportControl assessment to manage all necessary inputs for such an assessment from all over the SPDX model.
```
```suggestion
The Operations assessment descriptions are patterns for business relevant assessments like an ExportControl assessment to manage all necessary inputs for such an assessment from all over the SPDX model.
```

Do we have other instance of "Assessment" as a proper noun?

goneall commented:
Discussed on the 9 Dec 2025 Tech call:
- No reservations about merging into the develop branch
- Don't want to loose the formatting / typographical changes suggested by @bact 
- Agreed to just commit in the changes then do the merge
- If anyone on the operations team has concerns about the commits, they can create a new PR to change things back after the merge

goneall committed:
General formatting improvements

On the 9 Dec 2025 tech call we agreed to commit these improvement changes before merging PR #1172.  If the operations team disagrees with any of these changes, please submit an additional PR

Co-authored-by: Arthit Suriyawongkul <arthit@gmail.com>
Signed-off-by: Gary O'Neall <gary@sourceauditor.com>

goneall committed:
Apply suggestions from code review

Co-authored-by: Arthit Suriyawongkul <arthit@gmail.com>
Signed-off-by: Gary O'Neall <gary@sourceauditor.com>

goneall reviewed (approved):
Reviewed and approved on the 9 Dec 2025 Tech call
